app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/LearningActivity.kt

```diff
@@ -2,35 +2,43 @@ package com.github.HumanLearning2021.HumanLearningApp.view
 
 import android.content.ClipData
 import android.content.ClipDescription
-import android.media.MediaPlayer
 import android.os.Bundle
 import android.util.Log
 import android.view.DragEvent
 import android.view.MotionEvent
 import android.view.View
 import android.widget.ImageView
-import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
 import androidx.lifecycle.lifecycleScope
-import com.github.HumanLearning2021.HumanLearningApp.BuildConfig
 import com.github.HumanLearning2021.HumanLearningApp.R
-import com.github.HumanLearning2021.HumanLearningApp.model.DummyDatabaseService
-import com.github.HumanLearning2021.HumanLearningApp.presenter.DummyUIPresenter
-import com.github.HumanLearning2021.HumanLearningApp.presenter.DummyLearningPresenter
+import com.github.HumanLearning2021.HumanLearningApp.model.Category
+import com.github.HumanLearning2021.HumanLearningApp.model.Dataset
+import com.github.HumanLearning2021.HumanLearningApp.presenter.LearningPresenter
 import kotlinx.coroutines.launch
 
 class LearningActivity : AppCompatActivity() {
 
-    private lateinit var learningPresenter: DummyLearningPresenter
+    private lateinit var learningPresenter: LearningPresenter
     private lateinit var learningMode: LearningMode
     private lateinit var audioFeedback: LearningAudioFeedback
+    private lateinit var dataset: Dataset
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_learning)
         learningMode =
             intent.getSerializableExtra(LearningSettingsActivity.EXTRA_LEARNING_MODE) as LearningMode
-        learningPresenter = DummyLearningPresenter(learningMode)
+        val maybeDataset =
+            intent.getParcelableExtra<Dataset>(LearningDatasetSelectionActivity.EXTRA_SELECTED_DATASET)
+        if (maybeDataset != null) {
+            dataset = maybeDataset
```

Comment:
question: the use of the dataset is only to provide the categories from which to retrieve (representative) pictures from? i.e., given the categories, we retrieve any picture from the database that falls into these categories? As an example, then, it wouldn't be possible to have 2 different datasets of forks. Is this just for the dummy, or is it gonna stick when using firebase. I understand this is is more something to discuss with Jonas and Louis, but this made me think of it.

Reply:
Why wouldn't it be possible to have 2 different datasets of forks? The datasets are unique in the db, the categories as well, and you can combine them how you want. Or maybe I didn't understand well enough?

Reply:
This is tightly linked to my task, you should be able to retrieve all datasets containing the category and not the other way around where you have to spawn the datasets and fetch the categories